File: 프로그래머스/2/42842. 카펫/카펫.py
def solution(brown, yellow):
    answer = []
    number = brown+yellow
    # 전체 약수를 구함
    for i in range(1, number+1):
        if number%i == 0:
            answer.append(i)

    for h in answer:
        w = number // h
        if (w-2)*(h-2) == yellow:
            return [w, h]

File: 프로그래머스/2/42842. 카펫/카펫.java
import java.util.*;

class Solution {
    public int[] solution(int brown, int yellow) {
        List<Integer> lst = new ArrayList<>();
        int[] answer = new int[2];
        int num = brown + yellow;
        
        for(int i=1; i<=num; i++) {
            if (num % i == 0)
                lst.add(i);
        }
    
        for(int h : lst) {
            int w = num / h;
            if((w-2) * (h-2) == yellow) {
                answer[0] = w;
                answer[1] = h;
                
                break;
            }
        }
        return answer;
    }
}